k2/csrc/ragged.cu

```diff
@@ -34,6 +34,25 @@ struct RowSplitsDiff {
     return tmp;
   }
 };
+
+/*
+A helper function used in RaggedShape3;
+  if both first and second are non-NULL, it will check if the context of them
+     compatible or not and return that context if compatible;
+  if one of them is NULL, returns the other one's context.
+ */
+static k2::ContextPtr GetContext(const k2::Array1<int32_t> *first,
+                                 const k2::Array1<int32_t> *second) {
+  K2_CHECK(first != nullptr || second != nullptr)
+      << "At least one of first and second must be non-NULL";
+  if (first == nullptr)
+    return first->Context();
```

Comment:
this line will result in segfault since `first` is NULL.

Reply:
oh, sorry,